utils/obsproc/Smos2Ioda.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include <iostream>
+#include <netcdf>    // NOLINT (using C API)
+#include <string>
+#include <vector>
+
+#include "eckit/config/LocalConfiguration.h"
+
+#include <Eigen/Dense>    // NOLINT
+
+#include "ioda/Group.h"
+#include "ioda/ObsGroup.h"
+
+#include "NetCDFToIodaConverter.h"
+
+namespace gdasapp {
+
+  class Smos2Ioda : public NetCDFToIodaConverter {
+   public:
+    explicit Smos2Ioda(const eckit::Configuration & fullConfig)
+    : NetCDFToIodaConverter(fullConfig) {
+      variable_ = "Salinity";
+    }
+
+    // Read netcdf file and populate iodaVars
+    gdasapp::IodaVars providerToIodaVars(const std::string fileName) final {
+      oops::Log::info() << "Processing files provided by SMOS" << std::endl;
+
+      // Open the NetCDF file in read-only mode
+      netCDF::NcFile ncFile(fileName, netCDF::NcFile::read);
+
+      // Get number of obs
+      int nobs = ncFile.getDim("n_grid_points").getSize();
+
+      // Set the int metadata names
+      // TODO(AFE): add other metadata in form of
+      // std::vector<std::string> intMetadataNames = {"pass", "cycle", "mission"};
+      std::vector<std::string> intMetadataNames = {};
+
+      // Set the float metadata name
+      // TODO(AFE): add other metadata in form of
+      // std::vector<std::string> floatMetadataNames = {"mdt"};
+      std::vector<std::string> floatMetadataNames = {};
+      // Create instance of iodaVars object
+      gdasapp::IodaVars iodaVars(nobs, floatMetadataNames, intMetadataNames);
+
+      float lat[iodaVars.location];  // NOLINT
+      ncFile.getVar("Latitude").getVar(lat);
+
+      float lon[iodaVars.location];  // NOLINT
+      ncFile.getVar("Longitude").getVar(lon);
+
+      float sss[iodaVars.location];  // NOLINT
+      ncFile.getVar("SSS_corr").getVar(sss);
+
+      float sss_error[iodaVars.location];  // NOLINT
+      ncFile.getVar("Sigma_SSS_corr").getVar(sss_error);
+
+      unsigned short sss_qc[iodaVars.location];  // NOLINT
+      ncFile.getVar("Dg_quality_SSS_corr").getVar(sss_qc);
```

Comment:
@guillaumevernieres Is there some reason not to do this?

Reply:
You can't pass an std::vector to getVar @AndrewEichmann-NOAA . Maybe we should have been using ```netcdfcpp.h```? 
Wouldn't ```ulimit -s unlimited``` fix our possible stack size issue @rmclaren ?

Reply:
@guillaumevernieres That will make the program run, yes. It does remove some safety's (for example early detection of infinite recursion). Like I said its generally better to avoid huge stack allocations in the first place, by allocating memory on the heap instead (thats what its for)... 

I believe you could use`std::vector` as follows:

```
      std::vector<float> lat(iodaVars.location);
      ncFile.getVar("Latitude").getVar(lat.data());
```

std::vector<>::data() gives you the pointer to the internal buffer.

Reply:
Done